### PR TITLE
day 58 progress

### PR DESCRIPTION
**Pull request by nayabatir1**

### CODE REVIEW

File: atir-nayab/deployment/nginx-dep.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+apiVersion: apps/v1beta2
```
Comment (kranurag7) [inline atir-nayab/deployment/nginx-dep.yml:1]:
```suggestion
apiVersion: apps/v1
```

File: atir-nayab/06-04-2022/deployment101.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
+> scenario: you deployed an app few months ago. Now you want to upgrade your app from v1 to v2.
+
+> _can you upgrade with zero downtime?_  
+> _can you upgrade sequentially one after another?_  
+> _can you pause and resume upgrade process?_  
+> _Rollback upgrade to previous stable release_
+
+# deployment
+
+a deployment cotroller provides declarative updates for pods and replicaSets.
```
Comment (kranurag7) [inline atir-nayab/06-04-2022/deployment101.md:10]:
```suggestion
a deployment controller provides declarative updates for pods and replicaSets.
```

File: atir-nayab/08-06-2022/scheduling.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+- the kubernetes scheduler is a core component of kubernetes: after a user or a controller creates a pod, the kubernetes scheduler, monitoring the object store for unassigned pods, will assign the pod to a node. Then, the kubelet, monitoring the object store for assingned pods, will execute the pod.
+
+# what is the scheduler for?
+
+![scheduler](../asset/scheduler.png)
+
+the kubernetes scheduler is in charge of scheduling pods onto nodes. basically it works like this:
+
+1. you create a pod
+1. he scheduler notices that the new pod you created doesn't have a node assigned to it.
```
Comment (kranurag7) [inline atir-nayab/08-06-2022/scheduling.md:12]:
```suggestion
1. The scheduler notices that the new pod you created doesn't have a node assigned to it.
```

File: atir-nayab/08-06-2022/scheduling.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+1. the scheduler assigns a node tothe pod
```
Comment (kranurag7) [inline atir-nayab/08-06-2022/scheduling.md:13]:
```suggestion
1. the scheduler assigns a node to the pod
```

File: atir-nayab/08-06-2022/scheduling.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+# what is node taints and tolerations?
+
+- this kubernetes feature allows users to mark a node (taint the node) so that no pods can be scheduled to it, unless a pod explicitly tolerates the taint.
+- when you taint a node, it is automatically excluded from pod scheduling. when the schedule runs the predicate tests on a tainted node, thay'll fail unless the pod has toleration for that node.
+- let assume new member joins the development team, writes a deployment for her application, but forgets to exclude the monitoring node from the target nodes? kubernetes administrators need a way to repel pods from nodes without having to modify every pod definition.
+
+# tolerations
+
+- toleration is a way of ignoring a taint during scheduling. Tolerations aren't applied to nodes, but rather the pods. if we apply a toleration to teh podSpec, we could 'tolarate' the slow disks on that node and still use it.
```
Comment (kranurag7) [inline atir-nayab/08-06-2022/scheduling.md:51]:
```suggestion
- toleration is a way of ignoring a taint during scheduling. Tolerations aren't applied to nodes, but rather the pods. if we apply a toleration to the podSpec, we could 'tolerate' the slow disks on that node and still use it.
```

**Review by kranurag7 — CHANGES_REQUESTED** [review]

File: atir-nayab/deployment/nginx-dep.yml (hunk 1 above)
Comment (nayabatir1) [reply atir-nayab/deployment/nginx-dep.yml:1]: It will be v1beta2

Comment (kranurag7) [reply atir-nayab/deployment/nginx-dep.yml:1]: for deployments we will the `apps/v1`. Anything I'm missing here? Please let me know.

**Comment by kranurag7** [comment]
@nayabatir1 apologies from my side for being late. You can always create another PR through a different branch.

**Comment by kranurag7** [comment]
And Yes I really enjoy reading your notes. Keep it up.

**Pull request merged**